### PR TITLE
Restore HCAL DQM to offline workflow

### PR DESCRIPTION
**Pull request by DryRun**
Fix for non-backwards-compatible FED-to-crate tables, as discussed in https://github.com/cms-sw/cmssw/pull/17657: offline workflows have been modified to use crateId instead of FED number. 

FEDs are still used for online workflows. 

This PR contains most of the other commits from this year, which were not merged pending a fix for the fed2crate issue.

### CODE REVIEW

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
I guess these tests got stuck...

**Comment by kpedro88** [comment]
@cmsbuild please test

File: DQM/HcalCommon/interface/ElectronicsQuantity.h [hunk 1]
```diff
@@ -311,6 +401,50 @@ namespace hcaldqm
...
+			}
+			virtual ~CrateQuantity() {}
+
+			virtual void setup(HcalElectronicsMap const * emap);
+			virtual void setup(std::vector<int> crates, CrateHashMap crateHashes);
+			virtual int getValue(HcalElectronicsId const&);
+			virtual uint32_t getBin(HcalElectronicsId const&);
+
+			virtual int nbins() {
+				std::cout << "[debug] CrateQuantity: returning nbins = " << _crates.size() << std::endl;
```
Comment (dmitrijus) [inline DQM/HcalCommon/interface/ElectronicsQuantity.h]: You probably did this for debugging purposes and forgot to disable afterwards.
Either way, cout should not be used.

**Review by dmitrijus — CHANGES_REQUESTED** [review]

File: DQM/HcalCommon/interface/ElectronicsQuantity.h (hunk 1 above)
Comment (DryRun) [inline DQM/HcalCommon/interface/ElectronicsQuantity.h]: Thanks for spotting this. I committed a fix removing this line.

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1
from various conversations I believe there is some consensus that splitting the "offline", "online" and "common" parts of this into separate analyzers would greatly simplify and clarify the code. Hopefully that can be done in the near future..

**Pull request merged**